include/aws/common/ref_count.h

```diff
@@ -0,0 +1,54 @@
+#ifndef AWS_COMMON_REFCOUNT_H
+#define AWS_COMMON_REFCOUNT_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/common/common.h>
+
+#include <aws/common/atomics.h>
+
+typedef void(aws_on_zero_ref_count_callback)(void *);
```

Comment:
Nitpick: only thing I might change is to call this aws_destructor_callback or something more obvious for casting, maybe?